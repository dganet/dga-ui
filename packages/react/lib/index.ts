import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}