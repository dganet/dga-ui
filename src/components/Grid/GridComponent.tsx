import { GridProvider, useGrid } from '@hooks/useGrid/GridProvider'
import React, { useEffect } from 'react'

import { Container, StyledRow, StyledCol } from './GridComponent.styles'
import { GridProps, RowProps } from './GridComponent.types'

const Grid = ({ colGap, rowGap, children }: GridProps) => {
    return (
        <GridProvider>
            <GridComponent colGap={colGap} rowGap={rowGap}>
                {children}
            </GridComponent>
        </GridProvider>
    )
}

const GridComponent = ({ colGap, rowGap, children }: GridProps) => {
    const { setColGap, setRowGap } = useGrid()

    useEffect(() => {
        if (colGap) setColGap(colGap)
        if (rowGap) setRowGap(rowGap)
    }, [colGap, rowGap])

    return <Container>{children}</Container>
}

const Row: React.FC = ({ children }) => {
    const { rowGap } = useGrid()
    return <StyledRow gap={rowGap}>{children}</StyledRow>
}

const Col: React.FC<RowProps> = ({ children, lg, md, xl, sm, xs }) => {
    const { colGap } = useGrid()
    return (
        <StyledCol gap={colGap} xs={xs} sm={sm} md={md} lg={lg} xl={xl}>
            {children}
        </StyledCol>
    )
}

export { Grid, Row, Col }
