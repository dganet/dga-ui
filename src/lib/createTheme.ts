import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
