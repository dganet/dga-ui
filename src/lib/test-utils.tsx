import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
