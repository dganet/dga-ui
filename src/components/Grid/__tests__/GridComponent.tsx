import { render, screen } from '../../../lib/test-utils'

import GridComponent from '../GridComponent'

describe('<GridComponent />', () => {
  it('should render the component', () => {
    const { container } = render(<GridComponent />)

    expect(screen.getByRole('heading', { name: /Grid/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
