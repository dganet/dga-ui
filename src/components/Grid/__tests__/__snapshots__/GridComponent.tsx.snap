// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB"
>
  <h1>
    Grid - 
    new component
  </h1>
</div>
`;
